### PR TITLE
Fixes Deadminning keeping the AntagHUD on

### LINKED ISSUE #25802: Deadminning doesn't disable antag HUD

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #25802

### CODE REVIEW

File: code/modules/admin/verbs/randomverbs.dm [hunk 1]
```diff
@@ -944,10 +943,15 @@ var/list/datum/outfit/custom_outfits = list() //Admin created outfits
 		var/datum/atom_hud/antag/H = G.ganghud
 		(adding_hud) ? H.add_hud_to(usr) : H.remove_hud_from(usr)
 
-	to_chat(usr, "You toggled your admin antag HUD [adding_hud ? "ON" : "OFF"].")
-	message_admins("[key_name_admin(usr)] toggled their admin antag HUD [adding_hud ? "ON" : "OFF"].")
-	log_admin("[key_name(usr)] toggled their admin antag HUD [adding_hud ? "ON" : "OFF"].")
-	feedback_add_details("admin_verb","TAH") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+	if(!silent)
+		to_chat(usr, "You toggled your admin antag HUD [adding_hud ? "ON" : "OFF"].")
```
Comment (coiax) [inline code/modules/admin/verbs/randomverbs.dm]: Silent should still generate a message to admins and the log, just not message the person?

Comment (Cyberboss) [inline code/modules/admin/verbs/randomverbs.dm]: They already get the deadmin message. The implication is now that the antag hud is automatically turned off

**Comment by bgobandit** [comment]
rip fun

**Comment by ExcessiveUseOfCobblestone** [comment]
shoot

**Comment by kevinz000** [comment]
badmins

**Comment by Haevacht** [comment]
Kev's power revealed?

File: code/modules/admin/verbs/randomverbs.dm (hunk 1 above)
Comment (lzimann) [inline code/modules/admin/verbs/randomverbs.dm]: Should just scrap this silent entirely, if not, it should at least log and message admins about it.

File: code/modules/admin/verbs/randomverbs.dm [hunk 2]
```diff
@@ -947,7 +946,12 @@ var/list/datum/outfit/custom_outfits = list() //Admin created outfits
 	to_chat(usr, "You toggled your admin antag HUD [adding_hud ? "ON" : "OFF"].")
 	message_admins("[key_name_admin(usr)] toggled their admin antag HUD [adding_hud ? "ON" : "OFF"].")
 	log_admin("[key_name(usr)] toggled their admin antag HUD [adding_hud ? "ON" : "OFF"].")
-	feedback_add_details("admin_verb","TAH") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+	if(!silent)
+		feedback_add_details("admin_verb","TAH") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+/client/proc/has_antag_hud()
+	var/datum/atom_hud/A = huds[ANTAG_HUD_TRAITOR]
+	return usr in A.hudusers
```
Comment (RemieRichards) [inline code/modules/admin/verbs/randomverbs.dm]: usr is always a mob, so in this case, it's client.mob, so use that instead please.

File: code/modules/admin/admin_verbs.dm [hunk 3]
```diff
@@ -659,6 +659,9 @@ var/list/admin_verbs_hideable = list(
 
 	if(!holder)
 		return
+	
+	if(has_antag_hud())
+		toggle_antag_hud(TRUE)
```
Comment (RemieRichards) [inline code/modules/admin/admin_verbs.dm]: this looks bad.
typically with a ```toggle_X()``` proc they have an ```override``` or ```force``` argument, that forces it into that state (eg: the proc defaults to toggling but can also be used for forcing on or off), except in this case you're turning the hud off!

Off? True? WHO DO I BELIEVE!?

**Comment by Qbopper** [comment]
do not merge i need my valids

File: code/modules/admin/admin_verbs.dm (hunk 3 above)
Comment (optimumtact) [inline code/modules/admin/admin_verbs.dm]: just type it as toggle_antag_hud(silent=TRUE)

**Comment by lzimann** [comment]
do the thing pls

**Pull request merged**